docs/upgrade/upgrade-7.17-8.x.asciidoc

```diff
@@ -0,0 +1,152 @@
+
+[float]
+=== Pre-upgrade steps
+
+To prepare for the upgrade process, follow these steps before you start:
+
+. Do a software version inventory across your entire Elastic platform, including {es}, {kib}, {agent}, {beats}, and {ls}. If you're managing your deployments using {ece-ref}/ece-upgrade.html[{ece}] (ECE) or {eck-ref}/k8s-upgrading-eck.html[{eck}] (ECK), you may need to upgrade the orchestrator first.
+
+. If you're running any of these at versions earlier than 7.17, you must first upgrade them all to the latest 7.17.x patch release before upgrading to 8.x. This enables you to use the {kib} Upgrade Assistant to upgrade to 8.x. 
+
+. Run the {kibana-ref}/upgrade-assistant.html[Upgrade Assistant]. Take note of any critical error messages, then work with your Elastic Support rep to resolve them.
++
+.Requirements
+[sidebar]
+--
+To run the Upgrade Assistant, you must have `superuser` cluster admin permissions.
+--
+
+. If you're not using {ref}/snapshots-take-snapshot.html#automate-snapshots-slm[{slm} (SLM)], you must set up, configure, and run at least one {ref}/snapshot-restore.html[snapshot]. Rollbacks require a snapshot and cause data loss without one. Snapshots are {ref}/snapshot-restore.html#how-snapshots-work[incremental] – depending on the cluster size and the input/output rate, the initial snapshot may take several hours to complete. If you are using SLM, {ref}/snapshots-take-snapshot.html#check-slm-history[check the SLM history] to ensure that snapshots are completing successfully.
+
+[float]
+=== Perform 8.x upgrade on Non-Prod (Optional) or Prod Deployment
+
+NOTE: We strongly recommend you perform the following steps on a non-prod test deployment first to address any potential issues before upgrading on a prod deployment. 
+
+. Back up all of your data to a {ref}/snapshots-take-snapshot.html[snapshot] – which is a backup of indices taken from a running cluster. 
+
+. We recommend you <<rules-api-export, export all your custom detection rules>> in the event there are issues with the detection engine after the upgrade.
+
+. Upgrade {es}. Refer to {stack-ref}/upgrading-elasticsearch.html[these upgrade instructions]. 
+** If you're using {ecloud}, we recommend upgrades with no downtime. Refer to {cloud}/ec-upgrade-deployment.html[these instructions].  
+** If you're using {ece} (ECE), refer to {ece-ref}/ece-upgrade-deployment.html[these instructions].  
+** If you're using {eck} (ECK), refer to {eck-ref}/k8s-upgrading-stack.html[these instructions]. 
+** If you're upgrading a self-orchestrated deployment, upgrade the data nodes tier-by-tier, in this order:
+... Frozen tier
+... Cold tier 
+... Warm tier
+... Hot tier 
+... Any other nodes not in a tier
+... All remaining nodes that are neither master-eligible nor data nodes
+... Master-eligible nodes
+
+. Upgrade {kib}. Refer to {stack-ref}/upgrading-kibana.html[these instructions].
++
+NOTE: If you're using {ecloud} or {ece}, this is already included in the {es} upgrade.
```

Comment:
@shainaraskas not sure what ECH refers to? But I can convert the `{ecloud}` variable to `Elastic Cloud Hosted` if that's compliant w/ terminology standards.

Reply:
ECH is Elastic Cloud on Heroku. 

The meaning is important here because elastic cloud hosted (fka ESS / elastic cloud service) doesn't cover elastic cloud on kubernetes or elastic cloud on heroku, but those all fall under the umbrella of "Elastic Cloud" (so does Elastic Cloud Enterprise [ECE]).

Reply:
I'm learning all kind of gems! Thanks for the info, updated to `Elastic Cloud Hosted`.